modules/nextflow/src/main/groovy/nextflow/trace/TraceObserver.groovy

```diff
@@ -131,4 +131,17 @@ trait TraceObserver {
      *      The destination path at `publishDir` folder.
      */
     void onFilePublish(Path destination){}
+
+    /**
+     * Method that is invoke when an output file is published
+     * into a `publishDir` folder.
+     *
+     * @param source
+     *      The source path at `workDir` folder.
+     * @param destination
+     *      The destination path at `publishDir` folder.
+     */
+    void onFilePublish(Path source, Path destination){
```

Comment:
Let's declare this 

```
void onFilePublish(Path destination, Path source)
```

so that the `source` can be considered an optional argument from the API design point of view

Reply:
Sounds good!